crates/core/c8y_api/src/smartrest/smartrest_serializer.rs

```diff
@@ -1,15 +1,76 @@
+use crate::smartrest::csv::fields_to_csv_string;
 use crate::smartrest::error::SmartRestSerializerError;
 use crate::smartrest::topic::C8yTopic;
-use csv::QuoteStyle;
-use csv::WriterBuilder;
+use csv::StringRecord;
 use mqtt_channel::Message;
+use serde::ser::SerializeSeq;
 use serde::Deserialize;
 use serde::Serialize;
 use serde::Serializer;
 
 pub type SmartRest = String;
 
-#[derive(Debug)]
+pub fn request_pending_operations() -> &'static str {
+    "500"
+}
+
+/// Generates a SmartREST message to set the provided operation to executing
+pub fn set_operation_executing(operation: impl C8yOperation) -> String {
```

Comment:
The main additional changes from the last review are things like this, which replace the Serialization structs with functions.

Reply:
Indeed, the intent is better conveyed compared to these SmartRest codes such as 501.